### PR TITLE
Allow passing metadata to `accountRegister` mutation

### PR DESCRIPTION
**Pull request by piotrgrundas**
I want to merge this change because I want to support passing public metadata to:
- accountUpdate
- accountRegister

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/account/mutations/account.py [hunk 1]
```diff
@@ -38,6 +39,9 @@ class AccountRegisterInput(graphene.InputObjectType):
     language_code = graphene.Argument(
         LanguageCodeEnum, required=False, description="User language code."
     )
+    metadata = graphene.List(
+        MetadataInput, description="User public metadata.", required=False
```
Comment (korycins) [inline saleor/graphql/account/mutations/account.py:43]: `graphene.NonNull(MetadataInput)`

File: saleor/graphql/account/mutations/account.py [hunk 2]
```diff
@@ -123,6 +130,9 @@ class AccountInput(graphene.InputObjectType):
...
```
Comment (korycins) [inline saleor/graphql/account/mutations/account.py:134]: Should be NonNull

File: saleor/graphql/account/tests/test_account.py [hunk 3]
```diff
@@ -1535,6 +1549,21 @@ def test_logged_customer_update_anonymous_user(api_client):
     assert_no_permission(response)
 
 
+def test_logged_customer_updates_metadata(user_api_client):
```
Comment (IKarbowiak) [inline saleor/graphql/account/tests/test_account.py:1552]: It will be nice to also have for updating metadata in `AccountRegister` mutation. :)

**Review by korycins — APPROVED** [review]

File: saleor/graphql/account/tests/test_account.py (hunk 3 above)
Comment (piotrgrundas) [reply saleor/graphql/account/tests/test_account.py:1552]: @IKarbowiak But there is 👀 . Test for accountRegister is updated to include that change.

Comment (IKarbowiak) [reply saleor/graphql/account/tests/test_account.py:1552]: True! Sorry :)

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — COMMENTED** [review]
The current implementation of update metadata after by `accountUpdate`  loses data previously stored in metadata. 
eg. 
I have a user with metadata:
```
{
    "key": "k1",
    "value": "v1"
},
{
    "key": "k2",
    "value": "v2"
}
```
I want to update the value of k2 and add a new mete item. I should send:
```
input:{metadata:[{key:"k3", value:"v3"},{key:"k2", value:"v22"}]}
```
The current result is:
```
{
    "key": "k2",
    "value": "v22"
},
{
    "key": "k3",
    "value": "v3"
}
```

IMO expected result is:
```
{
    "key": "k1",
    "value": "v1"
},
{
    "key": "k2",
    "value": "v22"
},
{
    "key": "k3",
    "value": "v3"
}
```
@maarcingebala can you confirm?

**Comment by piotrgrundas** [comment]
@fowczarek I've used same mechanism as in dashboard. 
Second thing - U won't be able to ever clear metadata items or particular one, using proposed way.

**Comment by fowczarek** [comment]
But in the current implementation, you can reset your metadata accidentally. To delete metadata you should send `deleteMetadata` mutation. 
`updateMetadata` also doesn't clear existing metada.

**Comment by korycins** [comment]
IMO, Update should not overwrite the previously stored fields in meta.
Also maybe I am missing something, but what is the difference between updating metadata by calling `UpdateMetadata` and `accountUpdate`? I understand that it is good to have a possibility to insert something in the metadata during the account registration, but not sure what is a benefit for extending `accountUpdate`

**Comment by piotrgrundas** [comment]
@korycins
1) I followed dashboard pattern, but I won't argue here,
2) because it raises race condition in our integrations, and we talked about it (with U even) when discussing needed saleor dependencies.

**Comment by korycins** [comment]
So can't you reverse the order of calling mutation? Like first call updateMetadata and then call update accountUpdate? 
We are going to separate events related to update metadata from standard object, maybe this could be helpful for you?

I don't know all your business cases and edge cases but I don't like the idea that we add action to mutation which will do the same thing as mutation already introduced. I would prefer to define the way for such cases as future projects can have similar issuse but related to different object, like Order etc.

**Comment by patrys** [comment]
Why do we want multiple ways to manage metadata?

**Comment by piotrgrundas** [comment]
As agreed with @patrys we will only allow it in `accountRegister`.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**